src/fuse_reduce.cpp

```diff
@@ -221,11 +224,19 @@ struct find_pointwise_reduce
         auto reduce = r.result;
         auto input  = r.instructions["pointwise"];
 
+        // A broadcast with output shape's rank not matching input wouldn't work.
+        // TODO:  can the same error happen in find_reduce_reduce or find_reduce_pointwise?
+        if(contains(r.instructions, "broadcast"))
+        {
+            if(r.instructions["broadcast"]->get_shape().ndim() != input->get_shape().ndim() or
+               r.instructions["final_broadcast"]->get_shape().ndim() != input->get_shape().ndim())
```

Comment:
There is no need to check "final_broadcast".

Reply:
These lines are all deleted anyway.